node/lib/roles.js

```diff
@@ -0,0 +1,77 @@
+const _ = require('lodash');
+const RequestError = require('./Errors');
+
+const roleValidation = async (req, res, next) => {
+    let requestArray = req.url.split('/');
+    requestArray = _.pullAll(requestArray, ['']);
+    if (requestArray.length !== 2) {
+        return next();
+    }
```

Comment:
This bit is a little confusing... I don't really understand what is happening here..

Reply:
Sorry I just realized I'm signed into my personal account.... This is Heron.

Reply:
Sorry, yeah, that is confusing. The middleware is meant to be used on endpoints for a specific resource, like POST /project/:project_id, so it's parsing the URL and then if the URL is shorter/longer than it's expecting, it skips the rest.